File: games_services/android/src/main/kotlin/com/abedalkareem/games_services/models/Method.kt
package com.abedalkareem.games_services.models

enum class Method {
  Unlock, Increment, SubmitScore, ShowLeaderboards, ShowAchievements,
  LoadAchievements, SignIn, GetAuthCode, GetPlayerHiResImage, GetPlayerScore,
  GetPlayerScoreObject, SaveGame, LoadGame, GetSavedGames, DeleteGame, LoadLeaderboardScores
}

fun Method.value(): String {
  return "$this".lowercase()
}

fun methodsFrom(string: String): Method? {
  return Method.values().firstOrNull { it.value() == string.lowercase() }
}

File: games_services/darwin/Classes/Models/Method.swift
import Foundation

enum Method: String {
  case unlock = "unlock"
  case submitScore = "submitScore"
  case showLeaderboards = "showLeaderboards"
  case showAchievements = "showAchievements"
  case getPlayerHiResImage = "getPlayerHiResImage"
  case getPlayerScore = "getPlayerScore"
  case getPlayerScoreObject = "getPlayerScoreObject"
  case showAccessPoint = "showAccessPoint"
  case hideAccessPoint = "hideAccessPoint"
  case signIn = "signIn"
  case saveGame = "saveGame"
  case loadGame = "loadGame"
  case getSavedGames = "getSavedGames"
  case deleteGame = "deleteGame"
  case loadAchievements = "loadAchievements"
  case resetAchievements = "resetAchievements"
  case loadLeaderboardScores = "loadLeaderboardScores"
}
